### PR TITLE
Create account service

### LINKED ISSUE #207: Create `AccountService`
### Description
Create `AccountService`

### Requirements
- project should be runnable
- creating user must be successful
- test all the dependent services

### Deliverables
- `AccountService` with the following actions
    - `CreateUser`
    - `GetUserByEmailAddress`
    - `VerificateUser`
- register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by botirjonabdumannonovv**
Closes #207

### CODE REVIEW

**Review by Xolmuratov02 — APPROVED** [review]

**Review by Xolmuratov02 — APPROVED** [review]
Honey Code

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if (asNoTracking)
+            query = query.AsNoTracking();
+
+        return await query.SingleOrDefaultAsync(cancellationToken);
+    }
+
+    public async ValueTask<bool> VerifyUserAsync(string emailAddress, bool asNoTracking = false, 
+        CancellationToken cancellationToken = default)
+    {
+        if (emailAddress is null)
```
Comment (Ilxom-22) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs:49]: Bu yerda ham isNullOrWhiteSpace bilan tekshiring.

**Review by Ilxom-22 — CHANGES_REQUESTED** [review]

**Review by Ilxom-22 — APPROVED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+using AirBnB.Infrastructure.Common.Validators;
+using AirBnB.Persistence.DataContexts;
+using AirBnB.Persistence.Repositories.Interfaces;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+
+namespace AirBnB.Infrastructure.Common.Identity.Services;
+
+public class AccountService(
+    IdentityDbContext dbContext,
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs:13]: Why we are using `DbContext` directly in high level service ?

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        if (string.IsNullOrWhiteSpace(emailAddress))
+            throw new ArgumentNullException(nameof(emailAddress),"Email address cannot be null or empty.");
+
+        var query = userRepository.Get(user => user.EmailAddress == emailAddress);
+
+        if (asNoTracking)
+            query = query.AsNoTracking();
+
+        return await query.SingleOrDefaultAsync(cancellationToken);
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs:43]: Change to `FirstOrDefaultAsync`

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        if (string.IsNullOrWhiteSpace(emailAddress))
+            throw new ArgumentNullException(nameof(emailAddress),"Email address cannot be null or empty.");
+        
+        var user = dbContext.Users.Where(user => user.EmailAddress == emailAddress);
+
+        if (asNoTracking)
+            user = user.AsNoTracking();
+
+        return await user.AnyAsync(cancellationToken);
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Identity/Services/AccountService.cs:57]: I think we should set `IsVerified` property to `true` here, no ?

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by Xolmuratov02 — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]

**Pull request closed without merging**